src/ORM/Connect/DBQueryBuilder.php

```diff
@@ -165,6 +167,43 @@ protected function buildUpdateQuery(SQLUpdate $query, array &$parameters)
         return $sql;
     }
 
+    /**
+     * Returns the WITH clauses ready for inserting into a query.
+     */
+    protected function buildWithFragment(SQLSelect $query, array &$parameters): string
+    {
+        $with = $query->getWith();
+        if (empty($with)) {
+            return '';
+        }
+
+        $nl = $this->getSeparator();
+        $clauses = [];
+
+        foreach ($with as $name => $bits) {
+            $clause = $bits['recursive'] ? 'RECURSIVE ' : '';
+            $clause .= Convert::symbol2sql($name);
+
+            if (!empty($bits['cte_fields'])) {
+                $cteFields = $bits['cte_fields'];
+                // Ensure all cte fields are escaped correctly
+                array_walk($cteFields, function (&$colName) {
+                    $colName = preg_match('/^".*"$/', $colName) ? $colName : Convert::symbol2sql($colName);
```

Comment:
Not sure about the regex - would "MyTable.MyColumn" be escaped wrong? (should be "MyTable"."MyColumn")?

Reply:
This isn't meant to fix when someone does it wrong - it's just meant to account for unescaped (`MyTable.MyColumn`) and escaped (`"MyTable"."MyColumn"`) inputs. The arguments passed in really _shouldn't_ be escaped, but there's an inconsistency between when the ORM expects and doesn't expect escaped arguments, so this accepts and handles both.